pkg/filesystem/directory.go

```diff
@@ -93,6 +95,8 @@ type Directory interface {
 	// Symlink is the equivalent of os.Symlink().
 	Symlink(oldName string, newName string) error
 
+	UtimesNano(name string, ts []unix.Timespec) error
```

Comment:
Using the `unix` package as part of this interface seems like an unwise idea. This API is supposed to be OS agnostic. It's also fine to have virtual implementations of it that are independent of any OS.

Maybe it makes more sense to provide a `Chtimes()` function, similar to [`os.Chtimes()`](https://golang.org/pkg/os/#Chtimes)?